week_3/node-typescript-init-main/src/controllers/PostController.ts

```diff
@@ -0,0 +1,98 @@
+import { Request, Response } from "express";
+import { PostCreateDto } from "../interfaces/post/PostCreateDto";
+import PostService from "../services/PostService";
+import { PostBaseResponseDto } from "../interfaces/common/PostBaseResponseDto";
+import statusCode from "../modules/statusCode";
+import util from "../modules/util";
+import message from "../modules/responseMessage";
+import { PostUpdateDto } from "../interfaces/post/PostUpdateDto";
+import { PostResponseDto } from "../interfaces/post/PostResponseDto";
+const { validationResult } = require('express-validator');
+
+
+const createPost = async (req: Request, res: Response) => {
+
+    const error = validationResult(req);
+    if (!error.isEmpty()) {
+        console.log(error);
+        return res.status(statusCode.BAD_REQUEST).send(util.fail(statusCode.BAD_REQUEST, message.BAD_REQUEST, res.json({ errors: error.array() })));
+    }
+
+
+    const post: PostCreateDto = req.body;
+
+    try{
+        const data: PostBaseResponseDto = await PostService.createPost(post);
+        res.status(statusCode.CREATED).send(util.success(statusCode.CREATED, message.CREATE_POST_SUCCESS, data));
+
+    } catch(error){
+        console.log(error)
+        res.status(statusCode.INTERNAL_SERVER_ERROR).send(util.fail(statusCode.INTERNAL_SERVER_ERROR, message.INTERNAL_SERVER_ERROR));
+
+    }
+
+}
+
+
+const updatePost = async (req: Request, res: Response) => {
+    const { postId } = req.params;
+    const post: PostUpdateDto = req.body;
+
+    try {
+        await PostService.updatePost(postId, post);
+        res.status(statusCode.NO_CONTENT).send(); // .send() 안해주면 postman에서 무한로딩 돌아감
+
+    } catch (error) {
+        console.log(error)
+        res.status(statusCode.INTERNAL_SERVER_ERROR).send(util.fail(statusCode.INTERNAL_SERVER_ERROR, message.INTERNAL_SERVER_ERROR));
+        
+    }
+
+
+}
+
+
+const findByPostId = async (req: Request, res: Response) => {
+    const { postId } = req.params;
+    
+    try {
+
+        const data: null | PostResponseDto = await PostService.findByPostId(postId);
+
+        if (!data){
+            res.status(statusCode.NOT_FOUND).send(util.fail(statusCode.NOT_FOUND, message.NOT_FOUND));
+        } else {
+            res.status(statusCode.OK).send(util.success(statusCode.OK, message.READ_POST_SUCCESS, data));
+        }
```

Comment:
if 와 else 를 사용하게 되면 스타일상의 문제뿐만 아니라, 완전히 반전되는 로직을 작성하게 되는 위험이 생기게 됩니다. Js를 조금 더 클린하게 작성하는 방법으로는 else를 지양하는 것입니다!

```suggestion
        if (!data){
            res.status(statusCode.NOT_FOUND).send(util.fail(statusCode.NOT_FOUND, message.NOT_FOUND));
        } 
        
        res.status(statusCode.OK).send(util.success(statusCode.OK, message.READ_POST_SUCCESS, data));
        
```

이런식으로 else를 지양하는 편이 좋을 것 같습니다 !
관련해서 읽어볼만한 [블로그 글](https://velog.io/@jangws/9.-Else-if%EC%99%80-Else%EB%A5%BC-%ED%94%BC%ED%95%98%EA%B3%A0-Early-Return%ED%95%98%EC%9E%90)입니다.

Reply:
앗 이제야 확인했네요ㅠㅠ 꼼꼼한 코리 감사합니다 블로그 글도 읽어보겠습니다!!